### PR TITLE
Goog module exports

### PR DESCRIPTION
**Pull request by Alex7Li**
Previously, you could import the library by the syntax goog.require('helper') and goog.require('plain') However, the word 'helper' and 'plain' are common so this commit changes the names: 'datalayer.helper.DataLayerHelper' and 'datalayer.plain'. It also changes the name of the file is_plain_object.js to plain.js for consistency. 

Before, private functions that were exported along with dataLayerHelper. The google style guide says
"Symbols are only exported if they are meant to be used outside the module. " https://google.github.io/styleguide/jsguide.html#es-module-exports

I could not get karma to work without exporting the functions.Thus the private functions have just been moved to a different file to make the code more modular.

Now, if someone imports our code like a module, they can't mess up all the functions. I was planning to write something up about how to import our code like a goog module (i.e. how we will do it in [measurement library](https://github.com/googleinterns/measurement-library)), but that kind of import is only for people who are currently using the closure compiler, and it depends on how it is configured. 

To make the code maximally accessible, I just explained how to import the dist folder. This is also consistent with how other libraries are currently using data layer helper.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +18,24 @@ This library provides the ability to process messages passed onto a dataLayer qu
 - [License](#license)
   
 
+## Quick Start
+First, install the package with your favorite package manager
+```bash
+npm install data-layer-helper --save-dev
+// or
+yarn add data-layer-helper --dev
```
Comment (kjgalvan) [inline README.md:26]: I would not make the assumption that DLH is a dev dependency - libraries like Measurement Library will need it as a regular dependency. Other tools may need it as a regular dependency too if they are taking advantage of things like the command API or listener.

File: README.md [hunk 2]
```diff
@@ -17,6 +18,24 @@ This library provides the ability to process messages passed onto a dataLayer qu
...
+```
+
+Next, require the code in your javascript
```
Comment (kjgalvan) [inline README.md:29]: Nit: I would change the wording to import not require as require() is a different API than the example below.

File: src/helper/helper.js [hunk 3]
```diff
@@ -44,16 +44,9 @@
...
-
-const {type, hasOwn, isPlainObject} = goog.require('plain');
-
-/**
- * @define {boolean} When true, potential code errors will be logged to the
- * console. To enable this, run yarn build-debug to compile the distribution
- * code.
- */
-const DLH_DEBUG = goog.define('DLH_DEBUG', false);
+goog.module('datalayerhelper.helper.DataLayerHelper');
```
Comment (durbanie) [inline src/helper/helper.js:47]: I would change the name of the file too (to either data-layer-helper.js or data_layer_helper.js). This is Google convention, and can potentially be done in a separate PR.

File: src/helper/utils.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+ * Applies the given method to the value in the dataLayer with the given key.
+ * If the method is a valid function of the value, the method will be applies
+ * with any arguments passed in.
+ *
+ * @param {!Array<*>} command The array containing the key with the
+ *     method to execute and optional arguments for the method.
+ * @param {!Object<*>} model The current dataLayer model.
+ * @package
+ */
+function processCommand(command, model) {
```
Comment (durbanie) [inline src/helper/utils.js:15]: I thought we planned on moving this back into DataLayerHelper as a private (possibly static) method?

File: src/helper/utils.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+const DLH_DEBUG = goog.define('DLH_DEBUG', false);
+
+/**
+ * Enum for choosing the level at which to log an error.
+ * @readonly
+ * @enum {number}
+ * @package
+ */
+const LogLevel = {
+  LOG: 1,
```
Comment (durbanie) [inline src/helper/utils.js:153]: I'd use "INFO" here.

File: src/helper/utils.js [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+      } else {
+        to[property] = fromProperty;
+      }
+    }
+  }
+  delete to['_clear'];
+}
+
+
+/**
```
Comment (durbanie) [inline src/helper/utils.js:139]: What do you think about extracting the following logging-specific code into it's own file (e.g. logging.js)?

File: src/helper/utils.js [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+};
+
+/**
+ * Log an error to the console if the debug distribution is in use.
+ *
+ * @param {string} toLog The error to log to the console in debug mode.
+ * @param {!LogLevel} logLevel The error to log to the console in debug mode.
+ * @package
+ */
+function logError(toLog, logLevel) {
```
Comment (durbanie) [inline src/helper/utils.js:165]: Since we have info/warning here, I'd just call this function "log".

File: src/helper/utils.js (hunk 4 above)
Comment (Alex7Li) [reply src/helper/utils.js:15]: OK! I moved it into the DataLayerHelper as a private method and changed the tests to use it indirectly.

File: src/helper/helper.js (hunk 3 above)
Comment (Alex7Li) [reply src/helper/helper.js:47]: Switched it to data_layer_helper.js (an error was thrown when I tried data-layer-helper).

File: src/helper/utils.js (hunk 6 above)
Comment (Alex7Li) [reply src/helper/utils.js:139]: Great idea

File: src/helper/utils.js (hunk 7 above)
Comment (Alex7Li) [reply src/helper/utils.js:165]: Sounds good! It probably won't be confused with the math function since it takes a string argument

File: src/helper/helper.js (hunk 3 above)
Comment (Alex7Li) [reply src/helper/helper.js]: I misunderstood this the first time to be namespaces. I made the namespaces upperCamelCase and the filename data_layer_helper.js
